app/main.py

```diff
@@ -254,14 +247,78 @@ def change_zoom_center(n_clicks=None):
     return define_map_zoom_center(n_clicks, alert_metadata)
 
 
+@app.callback(
+    [Output('img_url', 'children'), Output('live_alert_header_btn', 'children'),
+     Output('live_alerts_marker', 'children')],
+    Input('interval-component', 'n_intervals'))
+def fetch_alert_status_metadata(n_intervals):
+    '''
+    This callback takes as input the interval-component which acts as a timer,
+    scheduling API metadata fetches and defining alert status
+    '''
+    # Fetching live alerts where is_acknowledged is False
+    response = api_client.get_ongoing_alerts().json()
+    all_alerts = pd.DataFrame(response)
+    live_alerts = all_alerts.loc[all_alerts['is_acknowledged'] == False]
+
+    # Defining alert status
+    if live_alerts.empty:
+        alert_status = 0
+        img_url = ""
+        return build_alerts_elements(img_url, alert_status, alert_metadata)
+    else:
+        alert_status = 1
+        # Fetching last alert
+        last_alert = live_alerts.loc[live_alerts['id'].idxmax()]
+
+        # Fetching last alert frame url
+        img_url = api_client.get_media_url(last_alert['media_id']).json()["url"]
+        return build_alerts_elements(img_url, alert_status, alert_metadata)
+
+
+'''
+To be uncommented for debug purposes
 @app.callback([Output('live_alert_header_btn', 'children'), Output('live_alerts_marker', 'children')],
-              [Input('alert_radio_button', 'value')])
-@cache.memoize()
-def define_alert_status(value=None):
+              [Input('alert_radio_button', 'value'), Input('interval-component', 'n_intervals')])
+def define_alert_status_debug(value=None, n_intervals=None):
+
+    This callback takes as input the alert_radio_button for debug purposes and defines the alert status
+    depending on the associated values
+
     if value is None:
-        value = 0
+        alert_status = 0
+    else:
+        alert_status = 1
+
+    return build_alerts_elements(alert_status, alert_metadata)
+'''
+
+
+@app.callback(Output('hp_alert_frame_metadata', 'children'),
+              Input('display_alert_frame_btn{}'.format(alert_id), 'n_clicks'), State('img_url', 'children'))
+def display_alert_frame_metadata(n_clicks_marker, img_url):
+    '''
+    This one detects the number of clicks the user made on an alert popup button.
+    If 1 click is made, the function returns the image of the corresponding alert.
+    '''
+    if (n_clicks_marker + 1) % 2 == 0:
+        return display_alerts_frames(n_clicks_marker, alert_metadata, img_url)
+    else:
+        return display_alerts_frames()
 
-    return build_alerts_elements(value, alert_metadata)
+
+@app.callback(
+    Output('interval-component', 'disabled'),
+    [Input("display_alert_frame_btn{}".format(alert_id), 'n_clicks')])
+def callback_func_start_stop_interval(n_clicks):
+    '''
+    This one detects the number of clicks the user made on a display_alert_frame_btn.
+    If 1 click is made, the function disables the interval component.
```

Comment:
I am not sure to understand the rationale behind this callback: if we stop refreshing the data fetched by the web-app as soon as the user has clicked on an alert, he might miss detections that are made afterwards, no? Or maybe this is only for the demo-day to ensure that the information about the alert sent by Matteo and the associated image, etc are stable?

Reply:
I'll be interested to get more insight on this topic, as I don't really get the goal with this callback

Reply:
> Or maybe this is only for the demo-day to ensure that the information about the alert sent by Matteo and the associated image, etc are stable?

Exactly ! As mentioned in notes, this is a workaround and should be changed in the future

Reply:
🤦‍♂️ I should have read more closely, sorry about that!